backend/testing/tester.py

```diff
@@ -136,23 +136,45 @@ def validate_message(self, body, message):
 
         # compare expected message with received one. exit on
         # mismatch.
-        for key in set(event.keys()).intersection(expected.keys()):
-            if "time" in key:
-                expected[key] += self.time_now
-            assert event[key] == expected[key] or (
-                isinstance(event[key], (list, set))
-                and set(event[key]) == set(expected[key])
-            ), (
-                'Test "{}" - Batch #{} - Type {}: Unexpected'
-                ' value for key "{}". Received: {}, Expected: {}'.format(
+        if isinstance(expected, dict):
+            for key in set(event.keys()).intersection(expected.keys()):
+                if "time" in key:
+                    expected[key] += self.time_now
+                assert event[key] == expected[key] or (
+                    isinstance(event[key], (list, set))
+                    and set(event[key]) == set(expected[key])
+                ), (
+                    'Test "{}" - Batch #{} - Type {}: Unexpected'
+                    ' value for key "{}". Received: {}, Expected: {}'.format(
+                        self.curr_test,
+                        self.curr_idx,
+                        message.delivery_info["routing_key"],
+                        key,
+                        event[key],
+                        expected[key],
+                    )
+                )
+        elif isinstance(expected, list):
+            correct = [True] * len(expected)
+            for idx, expected_item in enumerate(expected):
+                for key in set(event.keys()).intersection(expected_item.keys()):
+                    if "time" in key:
+                        expected_item[key] += self.time_now
+                    if event[key] != expected_item[key] or (
+                        isinstance(event[key], (list, set))
+                        and set(event[key]) != set(expected_item[key])
+                    ):
+                        correct[idx] = False
+                        break
+            if not any(correct):
```

Comment:
if one of them is correct though and the other failed, the test should fail right?

Reply:
so all of them need to be actually correct for the test to pass, not just one

Reply:
but you receive one at a time...

Reply:
one at a time as expected in the list. If even one fails, the test should fail.